subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt

```diff
@@ -1105,13 +1105,21 @@ data class PricingPhase(
     val billingPeriod: String,
 
 ) {
-    internal fun getBillingPeriodInDays(): Int? =
-        when (billingPeriod) {
-            "P1W" -> 7
-            "P1M" -> 30
-            "P1Y" -> 365
+    internal fun getBillingPeriodInDays(): Int? {
+        val regex = Regex("""P(\d+)([DWMY])""")
+        val match = regex.matchEntire(billingPeriod) ?: return null
+
+        val (amountStr, unit) = match.destructured
+        val amount = amountStr.toIntOrNull() ?: return null
+
+        return when (unit) {
+            "D" -> amount
+            "W" -> amount * 7
+            "M" -> amount * 30
+            "Y" -> amount * 365
             else -> null
         }
```

Comment:
According to the billing library [documentation](https://developer.android.com/reference/com/android/billingclient/api/ProductDetails.PricingPhase#getBillingPeriod()), `billingPeriod` is using standard ISO 8601 format. Java stdlib has a parser for that, I think it would be safer to use it instead of implementing a custom one.

```suggestion
       return try {
            val period = Period.parse(billingPeriod)
            return period.days + period.months * 30 + period.years * 365
        } catch (e: DateTimeParseException) {
            logcat { "Subs: Failed to parse billing period \"$billingPeriod\": ${e.asLog()}" }
            null
        }
```